helm-v3/006-repositories.md

```diff
@@ -1,13 +1,28 @@
 # Repositories
 
+A new API version for chart repositories (**v2**) will be introduced to provide new functionality address issues of scale (see [#3557](https://github.com/kubernetes/helm/issues/3557)).
+
 The following changes will be made to the repository subsystem:
 
-- The `helm serve` command will be removed
+- v2 will be the new default
+- v1 will be supported in its current state, but will not benefit from new functionality
+- The `helm search` command will become a server-side operation for v2 repos
+- No reliance on local index for `helm fetch` `helm inspect` `helm install` etc for v2 repos
```

Comment:
How is this a change from today? You can use `helm fetch` `helm inspect` `helm install` without adding a chart repository right now.

Reply:
What I mean by this is, if you are running a command like
`helm install  <repo>/<chart> --version=0.1.0`, there is some basic path spec that says that this equates to `GET <protocol>://<repo>/charts/chart/0.1.0`

This is so we do not have to rely on loading the entire index yaml containing all chart URLs to download a single package.

Reply:
@jdolitsky I think this sort of rigid layout spec can be very beneficial. By having a well defined layout required by the client and by making heavy load operations, like search, being optional and performed on the server side, we could actually make the whole indexing optional. This would not only facilitate the usage of pure object storage, but also promote the advantages of using a smart repository solution when needed.